plugin/src/main/java/org/opensearch/ml/action/profile/MLTaskProfileNodeResponse.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.action.profile;
+
+import org.opensearch.action.support.nodes.BaseNodeResponse;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContentFragment;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.ml.common.MLTask;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+
+public class MLTaskProfileNodeResponse extends BaseNodeResponse implements ToXContentFragment {
+    /**
+     * Node level MLTasks.
+     */
+    private Map<String, MLTask> mlNodeTasks;
+    /**
+     * Constructor
+     *
+     * @param in StreamInput
+     * @throws IOException throws an IO exception if the StreamInput cannot be reML from
+     */
+    public MLTaskProfileNodeResponse(StreamInput in) throws IOException {
+        super(in);
+        if (in.readBoolean()) {
+            this.mlNodeTasks = in.readMap(StreamInput::readString, MLTask::new);
+        }
+    }
+
+    public MLTaskProfileNodeResponse(DiscoveryNode node, Map<String, MLTask> nodeTasks) {
```

Comment:
Map<String, MLTask> --> is the string for node task id? I hope we aren't expecting readers to have intensive knowledge of this package.

Reply:
I added more comments in the next revision. The key is the taskId and the value is the MLTask that we want to retrieve in the memory to the users.